src/overlay/ItemPeerQueue.cpp

```diff
@@ -0,0 +1,60 @@
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "overlay/ItemPeerQueue.h"
+
+namespace stellar
+{
+
+void
+ItemPeerQueue::addKnowing(Peer::pointer peer)
```

Comment:
I think this class makes separate of concepts better. If we change strategy on how we query peer, we will have to change or remove this class, but merging it back into ItemFetchQueue does not seem as a good idea, as it would put again too much responsibility into one class.